components/push/ffi/src/lib.rs

```diff
@@ -139,10 +139,16 @@ pub extern "C" fn push_update(handle: u64, new_token: FfiStr<'_>, error: &mut Ex
 // verify connection using channel list
 // Returns a bool indicating if channel_ids should resubscribe.
 #[no_mangle]
-pub extern "C" fn push_verify_connection(handle: u64, error: &mut ExternError) -> u8 {
+pub extern "C" fn push_verify_connection(handle: u64, error: &mut ExternError) -> ByteBuffer {
     log::debug!("push_verify");
-    MANAGER.call_with_result_mut(error, handle, |mgr| -> Result<_> {
-        mgr.verify_connection()
+    MANAGER.call_with_result_mut(error, handle, |mgr| {
+        if !mgr.verify_connection()? {
```

Comment:
Can we move the `if !verify_connection() then regenerate_endpoints()` logic into it's own method in the manager? We try to put minimal logic in the ffi lib files.